aws-logs-destination/src/test/java/software/amazon/logs/destination/CreateHandlerTest.java

```diff
@@ -283,5 +286,172 @@ public void handleRequest_Should_ReturnFailureProgressEvent_When_DestinationRead
+                .thenReturn(DescribeDestinationsResponse.builder()
+                        .destinations(Collections.emptyList())
+                        .build())
+                .thenReturn(describeResponse);
+
+        final PutDestinationResponse putDestinationResponse = PutDestinationResponse.builder()
+                .destination(destination_without_policy)
+                .build();
+
+        final ResourceHandlerRequest<ResourceModel> request =
+                getDefaultRequestBuilder().desiredResourceState(testResourceModel)
+                        .build();
+
+        request.getDesiredResourceState().setDestinationPolicy(null);
+
+        Mockito.when(proxyClient.client()
+                .putDestination(ArgumentMatchers.any(PutDestinationRequest.class)))
+                .thenReturn(putDestinationResponse);
+
+        final ProgressEvent<ResourceModel, CallbackContext> response =
+                handler.handleRequest(proxy, request, new CallbackContext(), proxyClient, logger);
+
+        assertThat(response).isNotNull();
+        assertThat(response.getStatus()).isEqualTo(OperationStatus.SUCCESS);
+        assertThat(response.getCallbackDelaySeconds()).isEqualTo(0);
+
+        assertThat(response.getResourceModel()).isEqualTo(request.getDesiredResourceState());
+        assertThat(response.getResourceModels()).isNull();
+        assertThat(response.getMessage()).isNull();
+        assertThat(response.getErrorCode()).isNull();
+    }
+
+    @Test
+    public void handleRequest_Should_ReturnFailureProgressEvent_When_DestinationIsFound_and_DestinationPolicyNotProvided() {
+        final DescribeDestinationsResponse describeResponse = DescribeDestinationsResponse.builder()
+                .destinations(destination_without_policy)
+                .build();
+
+        Mockito.when(proxyClient.client()
+                .describeDestinations(any(DescribeDestinationsRequest.class)))
+                .thenReturn(describeResponse);
+
+        final ResourceHandlerRequest<ResourceModel> request =
+                getDefaultRequestBuilder().desiredResourceState(testResourceModel)
+                        .build();
+
+        request.getDesiredResourceState().setDestinationPolicy(null);
+
+        final ProgressEvent<ResourceModel, CallbackContext> response =
+                handler.handleRequest(proxy, request, new CallbackContext(), proxyClient, logger);
+
+        assertThat(response).isNotNull();
+        assertThat(response.getStatus()).isEqualTo(OperationStatus.FAILED);
+        assertThat(response.getErrorCode()).isEqualTo(HandlerErrorCode.AlreadyExists);
+    }
+
+    @Test
+    public void handleRequest_Should_ThrowCfnResourceConflictException_When_PutOperationIsAborted_and_DestinationPolicyNotProvided() {
+        final DescribeDestinationsResponse describeResponse = DescribeDestinationsResponse.builder()
+                .destinations(destination_without_policy)
```

Comment:
do we care about whether or not DescribeDestination returns a destination with a policy? it seems like we throw "ResourceNotFoundException" on line 440 anyways.

Reply:
agreed